src/TrifanHarness.cpp

```diff
@@ -73,6 +73,23 @@ int main()
             // probably should check arm/disarm here and abort if
             // disarmed
             ctrl->updateMotors(3500);
+            // if commanded to end takeoff, set motors to equillibrium
+            std::string takeoff_command;
+            while(true)
+            {
+              std::cout << "Enter 'end' to end takeoff operation.\n";
+              std::getline(std::cin, takeoff_command);
+              if (takeoff_command == "end")
+              {
+                ctrl->updateMotors(3000);
+                std::cout << "Takeoff operation complete.\n";
+                break;
+              }
+              else
+              {
+                std::cout << "Invalid command.\n";
+              }
+            }
```

Comment:
this totally works, I like the idea;  however, I notice this means that *until* the user presses "end" the actually can't do anything else.  Not even check the flight status to see how high they are yet.  Maybe instead we should have a "hover" command that's part of the top-level command tree that sets the motors to 3000 so that the user can monitor the status or do whatever else they want, and from any other state if they just want to snap to a "stable altitude" configuration that would be possible.  What do you think?  Another apporach would be to track the "current_state" as a new state item on the FlightController, and if the current state is "takeoff" you could add a new command to the command tree like "finish_takeoff" or "hover" or whatever, but once they LEAVE the takeoff state it's no longer available.  That's a little more complicated.  Either way I think locking them into the "end" command might be a bit too restrictive.

Reply:
Or alternatively your idea of taking off until you hit a fixed altitude is plausible.  You could even let them specify it as a parameter to the takeoff command like "takeoff 850" to climb to 850 feet.

Reply:
@evizitei Thanks for the feedback. I agree; I think it's too limiting as is. I really like the idea of specifying an altitude parameter because I do think the user needs to have some control over how high the aircraft will climb during takeoff.